src/s1ap/handlers/s1setup.c

```diff
@@ -21,132 +23,248 @@
 #include "sctp_conn.h"
 #include "s1ap_structs.h"
 #include "s1ap_msg_codes.h"
+#include "ProtocolIE-ID.h"
+#include "ProtocolIE-Field.h"
+#include "SupportedTAs-Item.h"
+#include "TAC.h"
+//void ut_ho(void *msg, int enb_fd);
+/******************************************************
+  S1_SETUP_RESPONSE handling
+******************************************************/
+int
+s1_setup_response(int enb_fd, struct PLMN *plmn)
+{
+    log_msg(LOG_DEBUG,"Process S1 Setup response.");
+	uint32_t length = 0;
+    uint8_t *buffer = NULL;
+	s1ap_config_t *s1ap_cfg = get_s1ap_config();
+    struct s1ap_common_req_Q_msg rsp_msg = {0};
+    rsp_msg.IE_type = S1AP_SETUP_RESPONSE;
+
+    memcpy(rsp_msg.mme_name, s1ap_cfg->mme_name, strlen(s1ap_cfg->mme_name));
+    rsp_msg.mme_code = s1ap_cfg->mme_code;
+	rsp_msg.mme_group_id = (s1ap_cfg->mme_group_id);
+    
+    struct PLMN local_plmn_id = {0};
+
+	local_plmn_id.idx[0] = plmn->idx[0];
+	local_plmn_id.idx[1] = plmn->idx[1];
+	local_plmn_id.idx[2] = plmn->idx[2];
+	log_msg(LOG_DEBUG,"Number of mnc digits %d \n", plmn->mnc_digits);
+    memcpy(&rsp_msg.mme_plmn_id, &local_plmn_id, sizeof(struct PLMN));
```

Comment:
struct PLMN is no more 3 bytes it has extra fields.... Just copy 3 bytes... just search the code for struct PLMN. You will figure out that we copy only 3 bytes and never complete struct PLMN.

@badhrinathpa @pingping-lin  this could be the issue with your handover test.

Reply:
@asreek  this needs resolution

Reply:
Fixed.
Couldn't find any other instances like this.